### PR TITLE
Add Configuration and Dependencies handling versions in sample apps

### PR DESCRIPTION
**Pull request by adasiewiczr**

### CODE REVIEW

File: buildSrc/src/main/kotlin/com/example/chattutorial/Dependencies.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.example.chattutorial
+
+private const val ANDROID_BUILD_TOOL_VERSION = "4.0.2"
```
Comment (zsmb13) [inline buildSrc/src/main/kotlin/com/example/chattutorial/Dependencies.kt]: This is actually the Android Gradle Plugin version, which is not the same as the build tools version.

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+plugins {
+    `kotlin-dsl`
+}
+
+repositories {
+    jcenter()
+}
```
Comment (zsmb13) [inline buildSrc/build.gradle.kts:7]: Do we need this, couldn't we just remove it?

File: samplejava/build.gradle [hunk 3]
```diff
@@ -38,16 +38,14 @@ android {
...
-    implementation 'com.github.bumptech.glide:glide:4.11.0'
-    annotationProcessor 'com.github.bumptech.glide:compiler:4.11.0'
+    implementation Dependencies.kotlinSTDLib
+
+    implementation Dependencies.lifecycleViewModel
+    implementation Dependencies.streamAndroid
+    implementation Dependencies.glide
+    annotationProcessor Dependencies.glideCompiler
 
     implementation fileTree(dir: "libs", include: ["*.jar"])
```
Comment (zsmb13) [inline samplejava/build.gradle]: We might as well remove this, if we're cleaning up

File: buildSrc/src/main/kotlin/com/example/chattutorial/Dependencies.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+private const val STREAM_ANDROID_VERSION = "4.3.0-beta-5"
+
+object Dependencies {
+    const val androidBuildToolGradlePlugin = "com.android.tools.build:gradle:$ANDROID_BUILD_TOOL_VERSION"
+    const val appCompat = "androidx.appcompat:appcompat:$APP_COMPAT_VERSION"
+    const val constraintLayout = "androidx.constraintlayout:constraintlayout:$CONSTRAINT_LAYOUT_VERSION"
+    const val glide = "com.github.bumptech.glide:glide:$GLIDE_VERSION"
+    const val glideCompiler = "com.github.bumptech.glide:compiler:$GLIDE_VERSION"
+    const val kotlinGradlePlugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:$KOTLIN_VERSION"
+    const val kotlinSTDLib = "org.jetbrains.kotlin:kotlin-stdlib:$KOTLIN_VERSION"
```
Comment (zsmb13) [inline buildSrc/src/main/kotlin/com/example/chattutorial/Dependencies.kt]: We usually format abbreviations into CamelCase as well, like in `minSdkVersion`. This should probably be `kotlinStdLib` instead.

File: samplekotlin/build.gradle [hunk 5]
```diff
@@ -39,13 +40,14 @@ android {
...
-    implementation 'androidx.appcompat:appcompat:1.2.0'
-    implementation 'androidx.constraintlayout:constraintlayout:2.0.1'
+    implementation Dependencies.kotlinSTDLib
+
+    implementation Dependencies.lifecycleViewModel
+    implementation Dependencies.streamAndroid
+    implementation Dependencies.glide
+    annotationProcessor Dependencies.glideCompiler
+
+    implementation fileTree(dir: "libs", include: ["*.jar"])
```
Comment (zsmb13) [inline samplekotlin/build.gradle]: Same as above, could be dropped

File: samplekotlin/build.gradle [hunk 6]
```diff
@@ -26,7 +27,7 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = JavaVersion.VERSION_1_8.toString()
```
Comment (zsmb13) [inline samplekotlin/build.gradle:30]: Neat, didn't know this was built-in 👍

File: buildSrc/build.gradle.kts (hunk 2 above)
Comment (adasiewiczr) [reply buildSrc/build.gradle.kts:7]: I think that It's needed to obtain `org.jetbrains.kotlin:kotlin-compiler-embeddable `

Comment (zsmb13) [reply buildSrc/build.gradle.kts:7]: Oh, right, we need it for the DSL plugin itself. I'd still swap it for `mavenCentral()`, but that's a nitpick 😄

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**